iot_config/src/session_key_service.rs

```diff
@@ -1,43 +1,175 @@
-use crate::{GrpcResult, GrpcStreamRequest, GrpcStreamResult};
+use crate::{
+    session_key::{self, SessionKeyFilter},
+    GrpcResult, GrpcStreamRequest, GrpcStreamResult, Settings,
+};
+use anyhow::Result;
+use file_store::traits::MsgVerify;
+use futures::stream::StreamExt;
+use helium_crypto::PublicKey;
 use helium_proto::services::iot_config::{
-    self, SessionKeyFilterGetReqV1, SessionKeyFilterListReqV1, SessionKeyFilterStreamReqV1,
-    SessionKeyFilterStreamResV1, SessionKeyFilterUpdateReqV1, SessionKeyFilterUpdateResV1,
-    SessionKeyFilterV1,
+    self, ActionV1, SessionKeyFilterGetReqV1, SessionKeyFilterListReqV1,
+    SessionKeyFilterStreamReqV1, SessionKeyFilterStreamResV1, SessionKeyFilterUpdateReqV1,
+    SessionKeyFilterUpdateResV1, SessionKeyFilterV1,
+};
+use sqlx::{Pool, Postgres};
+use tokio::{
+    pin,
+    sync::broadcast::{Receiver, Sender},
 };
-use tonic::Request;
+use tonic::{Request, Response, Status};
+
+pub struct SessionKeyFilterService {
+    admin_pubkey: PublicKey,
+    pool: Pool<Postgres>,
+    update_channel: Sender<SessionKeyFilterStreamResV1>,
+}
+
+impl SessionKeyFilterService {
+    pub async fn new(settings: &Settings) -> Result<Self> {
+        let (update_tx, _) = tokio::sync::broadcast::channel(128);
+        Ok(Self {
+            admin_pubkey: settings.admin_pubkey()?,
+            pool: settings.database.connect(10).await?,
+            update_channel: update_tx,
+        })
+    }
+
+    fn subscribe_to_session_keys(&self) -> Receiver<SessionKeyFilterStreamResV1> {
+        self.update_channel.subscribe()
+    }
 
-pub struct SessionKeyFilterService {}
+    fn clone_update_channel(&self) -> Sender<SessionKeyFilterStreamResV1> {
+        self.update_channel.clone()
+    }
+
+    fn verify_admin_signature<R>(&self, request: R) -> Result<R, Status>
+    where
+        R: MsgVerify,
+    {
+        request
+            .verify(&self.admin_pubkey)
+            .map_err(|_| Status::permission_denied("invalid admin signature"))?;
+        Ok(request)
+    }
+}
 
 #[tonic::async_trait]
 impl iot_config::SessionKeyFilter for SessionKeyFilterService {
     type listStream = GrpcStreamResult<SessionKeyFilterV1>;
     async fn list(
         &self,
-        _request: Request<SessionKeyFilterListReqV1>,
+        request: Request<SessionKeyFilterListReqV1>,
     ) -> GrpcResult<Self::listStream> {
-        unimplemented!();
+        let request = request.into_inner();
+
+        let filters = session_key::list_for_oui(request.oui, &self.pool)
+            .await
+            .map_err(|_| Status::internal("list session key filters failed"))?;
+
+        let (tx, rx) = tokio::sync::mpsc::channel(20);
```

Comment:
nit: add a const for the buffer size, used in multiple places so be simpler to bump size